### ISSUE TITLE
Accordion does not work well in combination with a screen reader

### ISSUE THREAD

**Issue by Jirodica** [ing-bank/lion#1503]
## Expected behavior

When you make use of this component with a SR, and a button is expanded you want to get focus on the expanded text when you make use of the arrow-keys before you get focus on the next button in the example in the simulation window.

## Actual Behavior

In the simulation window, if you turn on the screen reader (VO or NVDA) and focus is on the first button, you'll be told that it's collapsed. If you then click space or enter, it will expand neatly. If you then want to navigate to the expanded text with the arrow keys, the focus is first placed on the next button (Laboriosam), which is not the intention. When you've finally focused on the extended text (using the arrow keys), you'll get a message that you're in a group, which I don't think is necessary, and navigating with the arrow keys again puts the focus on the text.

https://webdemos.feature-testing.ing.net/p00019-ing-web-v2/components/content/accordion/overview/index.html

## Additional context

Accordion: Overview / version: ?

**Comment by daKmoR** [comment]
We actually on purpose opted out of this behavior for this rational

> Focusable elements should be interactive. Contents themselves do not offer any interactivity. If there is a button or a form inside the tab panel then these elements get focused directly.

see
https://lion-web.netlify.app/components/content/accordion/reference/#contents-are-not-focusable

would you agree with this rational? 🤔

@team maybe we should place this somewhere more prominent? 🤔

**Comment by Jirodica** [comment]
I don't agree with you on this one. The moment you click on the button, in this case, the hidden content is shown. People who use a screen reader want to see (hear) this content immediately, just like people who can see it directly.
SR users use the arrow keys to navigate through the non-interactive content.

**Comment by gerjanvangeest** [comment]
In case there is a button or link inside the content it is also later in the DOM tree then all other accordion invokers. Only moved around via css (order).

![image](https://user-images.githubusercontent.com/10164421/134465147-16e2ad98-d867-482c-a293-a72ed7ebb8bc.png)

So first all invokers get focus before the anchor inside the first panel gets it.

**Comment by Jirodica** [comment]
This is not how an SR user should experience this, if he or she clicks on a collapsed button, you have to come to the relevant content immediately after opening it. It's very simple, offer everyone the same experience. In addition, this is not really an accordian but 2 collapsed buttons stuck together. With an accordion, only 1 button can be expanded.

**Comment by daKmoR** [comment]
> With an accordion, only 1 button can be expanded.

from Wikipedia https://en.wikipedia.org/wiki/Accordion_(GUI)

The accordion is a graphical control element comprising a vertically stacked list of items, such as labels or thumbnails. Each item can be "expanded" or "collapsed" to reveal the content associated with that item. There can be zero expanded items, exactly one, or more than one item expanded at a time, depending on the configuration.

**Comment by erikkroes** [comment]
My impression is that there are some things being mixed up here.

The original issue is about the keyboard shortcut (down/right arrow) "clashing" with a screen reader shortcut. I think we can agree that's not what we want. 

I don't agree with moving the focus towards the content. It suggests we move the focus for every element with `aria-expanded`. By default, I prefer not "touching" the focus of a user. We don't know if all the content is relevant to the user. Let them decide. That's why a screen reader has the option to browse into the content, but also to jump to the next heading/link/etc. It's up to the user. They -can- percieve the content directly, by browsing to it.
Feel free to contact me and we can talk about this further, but I think it distracts from the original issue.

**Comment by erikkroes** [comment]
Trying to rephrase the original issue. Let me know if anything is unclear or plain wrong.


## Expected behavior

When you use a screen reader to activate the `<button>`, a new section appears. On a screen reader like VoiceOver, you can use the arrow keys to navigate to this new section (browse), so it gets announced.

## Actual Behavior

When using the arrow keys, the focus moves to the next `<button>` of the component. The new section is skipped.

**Comment by daKmoR** [comment]
if I look at https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html
this is behaving the same way? 
e.g. tab to accordion "heading1" => arrow down goes to "heading2"... if you want to go to the content you need to use "tab"

so are we saying that this wai aria practice example is wrong and we want to deviate? 🤔

**Comment by erikkroes** [comment]
It would be wrong to simply copy the wai aria practices. They're an ideal implementation, but not what's supported by all browsers and technologies. The big difference here is probably that `aria-controls` is used. This is meant to be a way for users to "jump" to an element. But, to put it mildly, that doesn't quite work: https://heydonworks.com/article/aria-controls-is-poop/
We're not a web components implementation of the wai aria practices. We have to deal with the implications of dealing with actual platforms.

**Comment by Jirodica** [comment]
See that this is still an issue, what is the planning to solve this?

**Comment by scottaohara** [comment]
It's fine for the up/down arrow keys to move to the expanding buttons.  that's what the APG example does.  The problem I'm seeing here is that when using VO + right arrow key, I expect to move to the content that is revealed after the first expanded button.  This is not happening with default VO setup, and instead focus moves past the revealed content to the next button in the accordion widget. 

As far as VO is concerned, if you remove the left/right arrow key functionality that would probably solve at least that issue.  Haven't tested this on a PC, so can't comment on any behavior there at this time.
